plumber/ppl/lib/ppl/repo_proxy_client.ex

```diff
@@ -95,14 +100,16 @@ defmodule Ppl.RepoProxyClient do
   end
 
   defp git_reference_from_ppl_req(ppl_req) do
-    branch_name = ppl_req.request_args |> Map.get("branch_name", "")
+    git_ref = ppl_req.request_args |> Map.get("git_reference", "")
 
-    if branch_name == "" do
-      raise "Provided an empty branch_name"
+    if git_ref == "" do
+      branch_name = ppl_req.request_args |> Map.get("branch_name", "")
+      "refs/heads/#{branch_name}"
```

Comment:
@DamjanBecirovic should we raise exception in case `branch_name` is empty

Reply:
The `branch_name` can not be empty since we have [a check](https://github.com/semaphoreio/semaphore/blob/main/plumber/ppl/lib/ppl/ppl_requests/model/ppl_requests.ex#L249) for that in Ecto changeset applied when we are creating a pipeline. All schedule requests that do not have branch_name will be rejjected due to that.